packages/api-plugin-promotions-discounts/src/handlers/getHighestCombination.js

```diff
@@ -0,0 +1,37 @@
+import _ from "lodash";
+
+/**
+ * @summary get the total discount on the cart
+ * @param {Object} cart - The cart to apply the promotion to
+ * @returns {Number} - The total discount on the cart
+ */
+function getTotalDiscountOnCart(cart) {
+  return cart.items.map((item) => (item.subtotal.discount || 0)).reduce((pv, cv) => pv + cv, 0);
+}
+
+/**
+ * @summary get the highest combination of promotions
+ * @param {Object} context - The application context
+ * @param {Object} cart - The cart to apply the promotion to
+ * @param {Object} combinations - The combinations to compare
+ * @returns {Promise<Array<Object>>} - The highest combination
+ */
+export default async function getHighestCombination(context, cart, combinations) {
+  const { promotions: { enhancers, utils } } = context;
+
+  const tasks = combinations.map(async (combinationPromotions) => {
+    let copiedCart = _.cloneDeep(cart);
+    for (const promo of combinationPromotions) {
+      // eslint-disable-next-line no-await-in-loop
+      const { affected, temporaryAffected } = await utils.actionHandler(context, copiedCart, promo);
```

Comment:
I assume the return values `affected` & `temporaryAffected` will be defined by the `actionHandler` funcs of child plugins (ex `packages/api-plugin-promotions-discounts/src/discountTypes/order/applyOrderDiscountToCart.js`). I have not reviewed them now and will review while merging feat/promotions.
Meanwhile, I could not find the field `temporaryAffected` being returned by any handler. only affected & reason were returned